spec/frameSpec.js

```diff
@@ -0,0 +1,145 @@
+describe('Frame', () => {
+  describe('constructor', () => {
+    it('following values are set upon initialisation', () => {
+      let frame = new Frame;
+      expect(frame.roll1).toEqual(null);
+      expect(frame.roll2).toEqual(null);
+      expect(frame.frameScore).toEqual(null);
+      expect(frame.strike).toEqual(false);
+      expect(frame.spare).toEqual(false);
+      expect(frame.frameFinished).toEqual(false);
+    })
+  })
+
+  describe('bowl', () => {
+    describe('first frame', () => {
+      let frame;
+
+      beforeEach(() => {
+        frame = new Frame;
+        spyOn(window, 'prompt').and.returnValue(1);
+        frame.bowl();
+        frame.bowl();
```

Comment:
This is interesting... you class reads the prompt directly? It's useful typically to separate the logic and the user interface. It also makes things easier to test.